Public/Src/Cache/MemoizationStore/Interfaces/Sessions/ContentHashList.cs

```diff
@@ -223,5 +225,56 @@ public string Serialize()
 
             return sb.ToString();
         }
+
+        /// <summary>
+        ///     Serialize whole value to a binary writer.
+        /// </summary>
+        public void Serialize(BuildXLWriter writer)

```

Comment:
>Serialize [](start = 20, length = 9)

I would move any serialization logic as an extension method in the assembly that uses it provided that the data is available. I don't think these types should be exposing a dependency on BuildXL.Utilities.